### PR TITLE
feat: data table for user view

### PR DESCRIPTION
**Pull request by laurensmertens**
|Q              |A  |
|---------------|---|
|Bug fix?       |No|
|New feature?   |Yes|	
|BC breaks?     |No|	
|Deprecations?  |No|	
|Fixed tickets  |No|	

Using DataTable for user view and created dynamic action functions

### CODE REVIEW

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -39,21 +40,48 @@ class UserController extends AppController
      */
     private $circleObject;
 
-    public function __construct(LoggerInterface $logger, FormRegistryInterface $formRegistry, RequestRuntime $requestRuntime, ?string $circleObject)
+    private $userService;
```
Comment (theus77) [inline src/Controller/UserController.php:43]:
```suggestion
    private UserService $userService;
```

File: src/Form/Data/TableColumn.php [hunk 2]
```diff
@@ -15,6 +15,8 @@ final class TableColumn
     private ?string $routeTarget = '_blank';
     private ?string $iconProperty = null;
     private ?bool $dateTimeProperty = null;
+    private ?bool $dataLinks = null;
+    private ?bool $arrayOfStrings = null;
```
Comment (theus77) [inline src/Form/Data/TableColumn.php:19]: Why null? Why not set them to false directly?

File: src/Form/Data/TableItemAction.php [hunk 3]
```diff
@@ -18,18 +18,21 @@ final class TableItemAction
     private $labelKey;
     /** @var string|null */
     private $messageKey;
+    /** @var bool */
+    private $dynamic;
```
Comment (theus77) [inline src/Form/Data/TableItemAction.php:22]:
```suggestion
    private bool $dynamic;
```

File: src/Resources/views/form/forms.html.twig [hunk 4]
```diff
@@ -50,7 +50,7 @@
                                     <td class="text-center">{{ loop.index }}</td>
                                     {% for column in table.columns %}
                                         {% set value = attribute(line.data, column.attribute)  %}
-                                        <td class="nowrap">
+                                        <td{% if column.arrayOfStrings != true %} class="nowrap"{% endif %}>
```
Comment (theus77) [inline src/Resources/views/form/forms.html.twig:53]:
```suggestion
                                        <td{% if not column.arrayOfStrings %} class="nowrap"{% endif %}>
```

File: src/Form/Data/TableColumn.php (hunk 2 above)
Comment (theus77) [inline src/Form/Data/TableColumn.php:19]:
```suggestion
    private bool $noWrap = true;
```

File: src/Resources/views/form/forms.html.twig [hunk 5]
```diff
@@ -77,20 +92,30 @@
...
+                                                {% endif %}
                                                 {% if action.post %}
-                                                    {% include '@EMSCore/elements/post-button.html.twig' with {
-                                                        'url': path(action.route, action.routeParameters|merge({(table.attributeName): line.data.id})),
-                                                        'message': action.messageKey,
-                                                        'label': action.labelKey,
-                                                        'itemLabel': attribute(line.data, table.labelAttribute),
-                                                        'icon': action.icon
-                                                    }%}
+                                                    {% if action.route != 'EMS_user_apikey' or 'ROLE_API' in line.data.roles  %}
```
Comment (theus77) [inline src/Resources/views/form/forms.html.twig:104]: Can you explain this line

Comment (laurensmertens) [reply src/Resources/views/form/forms.html.twig:104]: @theus77 I don't want the post-button to show if it concerns the "Generate API Key" button, unless if the user has the role ROLE_API.

File: src/Form/Data/TableColumn.php (hunk 2 above)
Comment (laurensmertens) [reply src/Form/Data/TableColumn.php:19]: I fixed the above remarks in another branch (upload-file-view)

**Pull request merged**